src/overlays/actors/ovl_En_Test7/z_en_test7.c

```diff
@@ -32,80 +46,943 @@ const ActorInit En_Test7_InitVars = {
+        this->unk_148.unk_10 -= 0x2EE0;
+    }
+}
+
+void func_80AF1CA0(EnTest7* this, GlobalContext* globalCtx) {
+    Vec3f sp34;
+
+    if (func_80183DE0(&this->unk_18CC)) {
+        func_80AF082C(this, func_80AF1E44);
+    }
+
+    if (this->unk_18CC.frameCtrl.unk_10 > 60.0f) {
+        func_80AF1B68(this, globalCtx);
+    }
+
+    if ((this->unk_18CC.frameCtrl.unk_10 > 20.0f) && !(this->unk_144 & 0x40)) {
+        this->unk_144 |= 0x40;
+        Audio_PlaySfxAtPos(&this->actor.projectedPos, NA_SE_PL_WARP_WING_CLOSE);
+    }
+
+    if (this->unk_18CC.frameCtrl.unk_10 > 42.0f) {
+        if (!(this->unk_144 & 0x80)) {
+            this->unk_144 |= 0x80;
+            Audio_PlaySfxAtPos(&this->actor.projectedPos, NA_SE_PL_WARP_WING_ROLL);
+        }
+
+        if (Rand_ZeroOne() < 0.3f) {
+            Camera* camera = Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(globalCtx->playerActorCsIds[8]));
+            f32 rand = Rand_ZeroOne();
+
+            sp34.x = ((camera->eye.x - this->actor.world.pos.x) * rand) + this->actor.world.pos.x;
+            sp34.y = ((camera->eye.y - this->actor.world.pos.y) * rand) + this->actor.world.pos.y;
+            sp34.z = ((camera->eye.z - this->actor.world.pos.z) * rand) + this->actor.world.pos.z;
+
+            func_80AF0C30(this->unk_15C, &sp34, 1);
+            this->unk_144 |= 8;
+        }
+    } else {
+        this->unk_144 |= 1;
+    }
+}
+
+void func_80AF1E44(EnTest7* this, GlobalContext* globalCtx) {
+    Vec3f sp34;
+    Camera* camera;
+    f32 rand;
+
+    func_80AF1B68(this, globalCtx);
+
+    if (Rand_ZeroOne() < 0.3f) {
+        camera = Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(globalCtx->playerActorCsIds[8]));
+        rand = Rand_ZeroOne();
+        sp34.x = ((camera->eye.x - this->actor.world.pos.x) * rand) + this->actor.world.pos.x;
+        sp34.y = ((camera->eye.y - this->actor.world.pos.y) * rand) + this->actor.world.pos.y;
+        sp34.z = ((camera->eye.z - this->actor.world.pos.z) * rand) + this->actor.world.pos.z;
+        func_80AF0C30(this->unk_15C, &sp34, 1);
+    }
+
+    Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
+    func_80AF0C30(this->unk_15C, &sp34, 1);
+    this->unk_18FC[1].y += 0x2EE0;
+}
+
+void func_80AF1F48(EnTest7* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 temp = this->unk_1E54 - 86;
+    f32 temp_f0 = temp / 10.0f;
+    Vec3f sp20;
+
+    this->unk_144 |= 0x10;
+
+    this->unk_148.unk_08 = (-0.15f * temp_f0) + 0.25f;
+    this->unk_148.unk_0C = (-0.3f * temp_f0) + 0.4f;
+    this->unk_148.unk_10 -= 0x2EE0;
+
+    this->unk_144 |= 4;
+
+    if (this->unk_1E54 >= 96) {
+        func_80AF082C(this, func_80AF2030);
+        this->unk_144 &= ~0x10;
+    }
+    Math_Vec3f_Copy(&sp20, &this->actor.world.pos);
+    func_80AF0C30(this->unk_15C, &sp20, 1);
+}
+
+void func_80AF2030(EnTest7* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 temp = this->unk_1E54 - 96;
+    f32 sp1C = 1.0f - (temp / 4.00000000000000001f);
```

Comment:
Out of curiosity does `/ 4` work

Reply:
Nope.

Reply:
`/ 4.0f`? No way is that precision visible to the hex.

Reply:
It's not, that's the point. here's the func if you want to play with it: https://decomp.me/scratch/yEaOV